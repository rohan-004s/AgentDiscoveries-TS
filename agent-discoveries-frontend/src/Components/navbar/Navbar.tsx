import './Navbar.styles.css'

interface NavArgs {
  imageUrl?: string
}

const Navbar: React.FC<NavArgs> = (props) => {
  var profileUrl = '/defaultProfile.png'
  if (props.imageUrl !== undefined && props.imageUrl.length > 0) {
    profileUrl = props.imageUrl
  }

  return (
    <nav id="navbar" className="navbar" data-testid="navbar">
      <form id="search-agent-network">
        <input
          type="search"
          id="search"
          name="search-network"
          placeholder="Search the agent network"
        />
      </form>
      <div className="items" id="navbar-items" data-testid="navbar-items">
        <a href="/home" data-testid="navbar-item">
          Homepage
        </a>
        <a href="/about" data-testid="navbar-item">
          About
        </a>
      </div>
      <div className="profile">
        <a href="/profile" data-testid="navbar-item">
          <img src={profileUrl} />
        </a>
      </div>
    </nav>
  )
}

export default Navbar
