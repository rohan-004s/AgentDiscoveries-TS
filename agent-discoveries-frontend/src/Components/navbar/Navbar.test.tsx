import Navbar from './Navbar'
import { cleanup, render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'

afterEach(() => {
  cleanup()
})

describe('Navbar', () => {
  it('Matches the snapshot', () => {
    const navbar = render(<Navbar imageUrl={''} />)
    expect(navbar.asFragment()).toMatchSnapshot()
  })

  it('Must render on page', () => {
    render(<Navbar imageUrl={''} />)
    const navbar = screen.getByTestId('navbar')
    expect(navbar).toBeInTheDocument()
  })

  test('Renders with all items', () => {
    render(<Navbar imageUrl={''} />)
    const navbarItems = screen.getAllByTestId('navbar-item')
    expect(navbarItems).toHaveLength(3)
  })
})
