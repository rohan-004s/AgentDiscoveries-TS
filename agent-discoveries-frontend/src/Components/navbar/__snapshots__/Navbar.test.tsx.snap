// Vitest Snapshot v1

exports[`Navbar > Matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    data-testid="navbar"
    id="navbar"
  >
    <form
      id="search-agent-network"
    >
      <input
        id="search"
        name="search-network"
        placeholder="Search the agent network"
        type="search"
      />
    </form>
    <div
      class="items"
      data-testid="navbar-items"
      id="navbar-items"
    >
      <a
        data-testid="navbar-item"
        href="/home"
      >
        Homepage
      </a>
      <a
        data-testid="navbar-item"
        href="/about"
      >
        About
      </a>
    </div>
    <div
      class="profile"
    >
      <a
        data-testid="navbar-item"
        href="/profile"
      >
        <img
          src="/defaultProfile.png"
        />
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
